sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py

```diff
@@ -20,7 +20,47 @@
 from ._helpers import _host_only, _is_tag, _strip_alg
 
 
-class ArtifactArchitecture(str, Enum, metaclass=CaseInsensitiveEnumMeta):
+class TypeInsensitiveEnumMeta(CaseInsensitiveEnumMeta):
+    """Extensible enum metaclass to trick extended enums into pretending to be the same type."""
+    def __instancecheck__(cls, other):
+        is_instance = super().__instancecheck__(other)
+        if not is_instance:
+            # Let's check if this is an instance of an extended enum. To do this we compare the enum member maps
+            # excluding the current attribute, along with the class name.
+            try:
+                unextended = {k: v for k, v in other._member_map_.items() if v != other.value}
+                return unextended == cls._member_map_ and cls.__name__ == other.__class__.__name__
+            except:  # pylint:disable=bare-except
+                pass
+        return is_instance
+
+
+class EnumBase(str, Enum, metaclass=TypeInsensitiveEnumMeta):
+    """Extensible enum base to allow for unrecognized strings to be returned in the payload without
+    breaking any existing code.
+    """
+    @classmethod
+    def _extended(cls, value):
+        try:
+            return cls(value)
+        except ValueError:
+            try:
+                if value in [None, ""] or value.isspace():
+                    return None
+                value = str(value)
+                enum_members = {m.name: m.value for m in cls}
+                # Enums will accept almost any name, so we'll just uppercase it and remove whitespace.
+                extended_name = "".join(value.upper().split())
+                enum_members[extended_name] = value
+                ExtendedEnum = EnumBase(cls.__name__, enum_members)
+                return ExtendedEnum(value)
+            except:  # pylint:disable=bare-except
+                # In the case that something went wrong, we don't want to raise in case it breaks
+                # deserialization, so just fallback to returning the initial value.
+                return value
```

Comment:
This should return a plain `str` in the worst case scenario - however I was unable to get it to fail.